### ISSUE TITLE
Copy/paste record order

### ISSUE THREAD

**Issue by jaginn** [mui-org/material-ui-x#3163]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When grid records are selected and copied to the clipboard (Ctrl+C), the order in which they are copied is the order in which they were selected, rather than the order in which they appear in the grid.

### Expected behavior 🤔

When copying records to the clipboard (Ctrl+C) they should be copied in the order they appear in the grid.

### Steps to reproduce 🕹

Steps:

1. Select records individually in an order different to what is displayed in the grid
2. Copy the records to the clipboard (Ctrl+C)
3. Paste the records into a text editor (Ctrl+V)

### Context 🔦

We allow users of the application to sort grid data in the order that is important to them. Copying records in a different order defeats the purpose of customising the grid sort order.

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  System:
    OS: macOS 11.6
  Binaries:
    Node: 14.17.6 - ~/.nvm/versions/node/v14.17.6/bin/node
    Yarn: Not Found
    npm: 6.14.15 - ~/.nvm/versions/node/v14.17.6/bin/npm
  Browsers:
    Chrome: 95.0.4638.69
    Edge: Not Found
    Firefox: Not Found
    Safari: 15.0
  npmPackages:
    @emotion/react: ^11.4.1 => 11.4.1 
    @emotion/styled: ^11.3.0 => 11.3.0 
    @mui/core:  5.0.0-alpha.49 
    @mui/icons-material: ^5.0.1 => 5.0.1 
    @mui/lab: ^5.0.0-alpha.50 => 5.0.0-alpha.50 
    @mui/material: ^5.0.2 => 5.0.2 
    @mui/private-theming:  5.0.1 
    @mui/styled-engine:  5.0.1 
    @mui/styles: ^5.0.1 => 5.0.1 
    @mui/system:  5.0.2 
    @mui/types:  7.0.0 
    @mui/utils:  5.0.1 
    @mui/x-data-grid-pro: ^5.0.0-beta.4 => 5.0.0-beta.4 
    @mui/x-license-pro:  5.0.0-beta.4 
    @types/react: ^16.14.15 => 16.14.15 
    react: ^17.0.2 => 17.0.2 
    react-dom: ^17.0.2 => 17.0.2 
    typescript: ^4.1.6 => 4.1.6 
```
</details>

### Order ID 💳 (optional)

31318

**Comment by oliviertassinari** [comment]
Please provide a **minimal** reproduction test case with the latest version. This would help a lot 👷 .
A live example would be perfect. [This **codesandbox.io** template](https://material-ui.com/r/issue-template) _may_ be a good starting point. Thank you!

**Comment by flaviendelangle** [comment]
You are using the `5.0.0-beta.4` and it should be fixed in `5.0.0-beta.5` :+1: (#2941)
